stix_shifter_modules/crowdstrike/stix_transmission/connector.py

```diff
@@ -31,45 +36,76 @@ def _handle_errors(self, response, return_obj):
         """
         response_code = response.code
         response_txt = response.read().decode('utf-8')
+        response_type = response.headers.get('Content-Type')
+        response_dict = {}
 
         if 200 <= response_code < 300:
             return_obj['success'] = True
             return_obj['data'] = response_txt
             return return_obj
-        elif ErrorResponder.is_plain_string(response_txt):
-            ErrorResponder.fill_error(return_obj, message=response_txt)
-            raise Exception(return_obj)
-        elif ErrorResponder.is_json_string(response_txt):
-            response_json = json.loads(response_txt)
-            ErrorResponder.fill_error(return_obj, response_json, ['reason'], connector=self.connector)
-            raise Exception(return_obj)
-        else:
-            raise Exception(return_obj)
+        elif response_code >= 400:
+            if response_type == 'application/json':
+                error_response = json.loads(response_txt)
+                response_dict['type'] = 'ValidationError'
+                response_dict['message'] = error_response['errors'][0]['message']
+                ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+                raise QueryException(return_obj)
+            elif response_type == 'text/html':
+                error = ConnectionError(f'Error connecting the datasource: {response_txt}')
```

Comment:
Would `reponse_txt` contain the name of the connector (crowdstrike) or should it be added to the message?

Reply:
it should be in the message when error message is constructed in the next function call.
`ErrorResponder.fill_error(return_obj, response_dict, error=error, connector=self.connector)`

Error message basically will look like this-
` "crowdstrike connector error => Connection error: Error connecting the datasource: <Returned errors from datasource>"`